docs/dev/content/implementation.rst

```diff
@@ -0,0 +1,116 @@
+Implementation Details and Workflows
+====================================
+
+To achieve using separated database for each channel and being able to switch channel dynamically, the following date structure and utility functions have been implemented.
+
+ContentDBRoutingMiddleware
+--------------------------
+
+This middleware will get used on every request sent to content app, and every response content app sends out.
+If a channel ID was included in the URL, It will ensure the appropriate content DB is used for the duration of the request. (Note: `set_active_content_database` is thread-local, so this shouldn't interfere with other parallel requests.)
+
+For example, this is how the client side dynamically request data from different channels:
+
+    >>> localhost:8000/api/content/<channel_1_id>/contentnode
+
+this will response with all the contentnode data stored in database channel_1_id.sqlite3
```

Comment:
"response" -> "respond"
"channel_1_id.sqlite3" -> "<channel_1_id>.sqlite3" (otherwise, it seems like a literal value, rather than a value to be filled in)